src/main/java/com/hermesworld/ais/galapagos/kafka/util/AclSupport.java

```diff
@@ -98,27 +92,99 @@ public Collection<AclBinding> getRequiredAclBindings(String environmentId, Appli
+                .collect(Collectors.toSet());
+    }
+
     private Collection<AclBinding> prefixAcls(String userName, ResourceType resourceType, String prefix,
-            List<AclOperation> ops) {
-        ResourcePattern pattern = new ResourcePattern(resourceType, prefix, PatternType.PREFIXED);
-        return ops.stream()
-                .map(op -> new AclBinding(pattern, new AccessControlEntry(userName, "*", op, AclPermissionType.ALLOW)))
+            List<AclOperationAndType> ops) {
+        return ops.stream().map(op -> op.toBinding(prefix, resourceType, PatternType.PREFIXED, userName))
                 .collect(Collectors.toList());
     }
 
-    private Collection<AclBinding> topicAcls(String userName, String topicName, List<AclOperation> ops) {
-        ResourcePattern pattern = new ResourcePattern(ResourceType.TOPIC, topicName, PatternType.LITERAL);
-        return ops.stream()
-                .map(op -> new AclBinding(pattern, new AccessControlEntry(userName, "*", op, AclPermissionType.ALLOW)))
+    private Collection<AclBinding> topicAcls(String userName, String topicName, List<AclOperationAndType> ops) {
+        return ops.stream().map(op -> op.toBinding(topicName, ResourceType.TOPIC, PatternType.LITERAL, userName))
                 .collect(Collectors.toList());
     }
 
-    private Collection<AclBinding> transactionAcls(String userName, String prefix) {
-        return Stream.of(AclOperation.DESCRIBE, AclOperation.WRITE)
-                .map(op -> new AclBinding(
-                        new ResourcePattern(ResourceType.TRANSACTIONAL_ID, prefix, PatternType.PREFIXED),
-                        new AccessControlEntry(userName, "*", op, AclPermissionType.ALLOW)))
-                .collect(Collectors.toSet());
+    private static class AclOperationAndType {
+
+        private final AclOperation operation;
+
+        private final AclPermissionType permissionType;
+
+        private AclOperationAndType(AclOperation operation, AclPermissionType permissionType) {
+            this.operation = operation;
+            this.permissionType = permissionType;
+        }
+
+        public AclBinding toBinding(String resourceName, ResourceType resourceType, PatternType patternType,
+                String principal) {
+            return new AclBinding(new ResourcePattern(resourceType, resourceName, patternType),
+                    new AccessControlEntry(principal, "*", operation, permissionType));
+        }
     }
 
+    private static class ResourcePatternAndPrincipal {
+
+        private final ResourcePattern resourcePattern;
+
+        private final String principal;
+
+        private ResourcePatternAndPrincipal(ResourcePattern resourcePattern, String principal) {
+            this.resourcePattern = resourcePattern;
+            this.principal = principal;
+        }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
Single-letter variables are not good for readability and I would prefer a change to "obj".